### ISSUE TITLE
Let me aggressively invite non-textsecure friends

### ISSUE THREAD

**Issue by holmesworcester** [WhisperSystems/TextSecure#3446]
I just set Textsecure as my default SMS app on Android. 

Every time I send an unconventional SMS, I'd like to invite these users to install Textsecure/Signal, because I want more of my friends to be able to have private conversations with me. I also want Textsecure to spread organically because I think that'd be good for the world!

I'd like a preference that would append short invite text and an invite link to every message provided there are enough characters, for example:

"Let's speak privately! SHORTLINK"

or "Get privacy: SHORTLINK"

or "Get this! SHORTLINK"

Ideally you'd experiment and figure out whatever text worked the best. For any user that sets their default SMS app as textsecure/signal, you should ask them if they'd like to turn this on.

**Comment by Thunderpig72** [comment]
+1.

My suggestion is "Texts for free: Shortlink"

**Comment by McLoo** [comment]
Selecting  invite from the menu is not good enough?

**Comment by lescheff** [comment]
My experience is that bugging people aggressively only makes it harder to convince them.

**Comment by watchpocket** [comment]
I absolutely cannot send "Get with it" to folks who are indifferent about the security of their texts.  That will only alienate them.  When it appears after I press the "Invite" menu item, I have to delete it and write in my own message. 

I'd like to suggest deleting it altogether and replacing it with something like: "A better texting app.  Don't believe me?  Give it a try: [SHORTLINK]" 

Selling folks on "better" is easier than selling them on "secure." Since they can never see anyone reading their texts, it isn't real to them.  I've often thought the only way I can interest people in security is to tell them I can read the texts they send to others, and then show those texts back to them, and say, "See?"  

Even if I knew how to do that, I'm not sure I'd really want to, but there's no question it would "secure" their attention to security.  For many, it would be the *only* way to get them thinking for real about their privacy.

**Comment by oaim** [comment]
I think it would be enough to append a message saying "Sent with Textsecure: [link]"
People would become curious and may give it a try

**Comment by holmesworcester** [comment]
Here’s how I think about it:

1) Many of your existing users are passionate private advocates and early adopters, like me. 

2) I would like to invite all my friends to use this and try it as their default texting app.

3) right now, there’s no great way for me to invite all my friends, or an easy way to invite all the people I message. Sending each person an invite individually is out of the question.

So… I think you need to think about how to get users inviting others.

You can test different messages and see what works best.

Also, the drawback of this approach is that it only works for people who’ve set textsecure as their default SMS app. 

Another approach would be to let people send messages to non-users, such that the user receives a notification that they have messages waiting but has to install textsecure to see their messages. I don’t have any sense of how hard this is to do. But this might be better a better approach, since it works for users who don’t have textsecure as their default app, and because there’s a really compelling reason for the user to install, in that moment. The workflow for somebody looking to start a private conversation becomes simpler, too.

–Holmes

**Comment by Thunderpig72** [comment]
I personally think the first message sent to someone should contain a Signal link. That way it won't get repetitive, but people will still hear about the app.

**Comment by holmesworcester** [comment]
I agree repetition is a problem. I’d probably opt for 1 out of 10 or something though.

–Holmes
